src/app/Api/Core/DAL/BaseRepository.cs

```diff
@@ -51,7 +51,7 @@ public async Task InsertManyAsync(IEnumerable<TDocument> documents)
         public async Task<TDocument> FindOneAsync(TFilter filter)
         {
             var result = await Collection.FindAsync(BuildFilterQuery(filter));
-            return result.SingleOrDefault();
+            return result.FirstOrDefault();
```

Comment:
I'd keep `SingleOrDefault()`  for additional validation. This is different from [MongoDB behavior](https://docs.mongodb.com/manual/reference/method/db.collection.findOne/). But usually `FindOne` is used when we need to get a single unique item from DB (often by Id). If it gets more than one item, either the filter or the data may be broken and it's better to get aware of this as soon as possible.